score-xml/src/main/java/io/cloudslang/content/xml/services/ValidateService.java

```diff
@@ -21,38 +23,61 @@
  * Created by markowis on 03/03/2016.
  */
 public class ValidateService {
-    public Map<String, String> execute(CommonInputs commonInputs, CustomInputs customInputs){
+    public Map<String, String> execute(CommonInputs commonInputs, CustomInputs customInputs) {
         Map<String, String> result = new HashMap<>();
 
         try {
-            String xmlDocument = commonInputs.getXmlDocument();
-            String xsdDocument = customInputs.getXsdDocument();
+            String xmlDocument;
+            String xsdDocument;
+            if (Constants.XML_PATH.equalsIgnoreCase(commonInputs.getXmlDocumentSource())) {
+                Document doc = XmlUtils.createDocumentFromFile(commonInputs.getXmlDocument(), commonInputs.getSecureProcessing());
+                xmlDocument = getStringValueOfDocument(doc);
+            } else if (Constants.XML_URL.equalsIgnoreCase(commonInputs.getXmlDocumentSource())) {
+                xmlDocument = XmlUtils.createXmlDocumentFromUrl(commonInputs);
+                xmlDocument = xmlDocument.replaceAll("<!DOCTYPE[^>]*>\n", "");
+            } else {
+                xmlDocument = commonInputs.getXmlDocument();
+            }
 
+            if (Constants.XSD_PATH.equalsIgnoreCase(customInputs.getXsdDocumentSource())) {
```

Comment:
the same checks are done for both XML_PATH and XSD_PATH, a better approach would make the code look cleaner. thanks.